astar.py

```diff
@@ -167,6 +168,18 @@ def get_node(x, y, nodes, parent):
     else:
         return None
 
+def get_nearest_coord(lat, lon, data):

```

Comment:
Looking at this again, I would probably re-name these arguments (at least the 'data' arg). It's unclear what `data` is supposed to be here.

Perhaps rename this to `nearest_coords_from_lat_lon`?